db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/post_provision.rb

```diff
@@ -1 +1,60 @@
+def find_vm_by_tag(tag)
+  tag = "/managed/deploy/#{tag}_#{$evm.root['automation_task'][:id]}"
+  $evm.vmdb(:vm).find_tagged_with(:any => tag, :ns => "*")
+end
 
+def assign_vms_to_deployment_nodes(masters, nodes)
+  deployment = $evm.vmdb(:container_deployment).find(
+    $evm.root['automation_task'].automation_request.options[:attrs][:deployment_id])
+  masters.each do |master|
+    deployment.assign_container_deployment_node(master.id, "master")
+  end
+  nodes.each do |node|
+    deployment.assign_container_deployment_node(node.id, "node")
+  end
+  $evm.root['deployment_master'] = deployment.roles_addresses("deployment_master")
+  $evm.root['inventory'] = deployment.regenerate_ansible_inventory
+  $evm.root['rhel_subscribe_inventory'] =  deployment.regenerate_ansible_subscription_inventory
+end
+
+def missing_subscription_fields?
+  $evm.root['rhsub_user'].nil? || $evm.root['rhsub_pass'].nil? || $evm.root['rhsub_sku'].nil?
+end
+
+def extract_ips(vms)
+  ips = []
+  vms.each do |vm|
+    ips << vm.hardware.ipaddresses.last unless vm.hardware.ipaddresses.empty?
+  end
+  ips
+end
+
+def wait_for_ip_adresses
+  $evm.root['state'] = "post_provision"
+  $evm.root['automation_task'].message = "Trying to recieve ips"
+  tagged_vms_masters = find_vm_by_tag("master")
+  tagged_vms_nodes = find_vm_by_tag("node")
+  masters_ips = extract_ips(tagged_vms_masters)
+  nodes_ips = extract_ips(tagged_vms_nodes)
+  if nodes_ips.count + masters_ips.count == tagged_vms_masters.count + tagged_vms_nodes.count
+    assign_vms_to_deployment_nodes(tagged_vms_masters, tagged_vms_nodes)
+    $evm.root['masters'] = masters_ips
+    $evm.root['nodes'] = nodes_ips
+    $evm.root['ae_result'] = 'ok'
+    $evm.root['container_deployment'] = $evm.vmdb(:container_deployment).find(
+      $evm.root['automation_task'].automation_request.options[:attrs][:deployment_id])
+  else
+    $evm.log(:info, "*********  Post-Provision waiting on ips ************")
+    $evm.root['ae_result']         = 'retry'
+    $evm.root['ae_retry_interval'] = '2.minute'
+    refresh_provider
+  end
+end
+
+def refresh_provider
+  deploy_on_provider = $evm.vmdb(:ext_management_system).find($evm.root['automation_task'].automation_request.options[:attrs][:provision_provider_id])
```

Comment:
@alongoldboim 
Can we add checks around this one of the chained commands could fail or return a nil